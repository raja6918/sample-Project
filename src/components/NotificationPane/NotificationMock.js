export const notificationResponse = [
  {
    notificationId: 1,
    scenarioName: 'July 2020 Production',
    solverRequestName: 'Combo Pilots',
    solverStatus: 'Done-success',
    timestamp: '27 June, 2020 at 19:02',
    unread: true,
  },
  {
    notificationId: 2,
    scenarioName: 'July 2020 Cabin Prelim for Forecast - a long scenario name',
    solverRequestName:
      'solver request This request has a very incredibly super max ultra pro long name',
    solverStatus: 'Done-failed',
    timestamp: '27 June, 2020 at 19:02',
    unread: true,
  },
  {
    notificationId: 3,
    scenarioName: 'July 2020 Production',
    solverRequestName: 'solver request This request has a medium length name',
    solverStatus: 'Done-success',
    timestamp: '27 June, 2020 at 19:02',
    unread: true,
  },
  {
    notificationId: 4,
    scenarioName: 'September 2020 Production',
    solverRequestName: 'Pilots 777',
    solverStatus: 'Done-success',
    timestamp: '27 June, 2020 at 19:02',
  },
  {
    notificationId: 5,
    scenarioName: 'Novermber 2020 Production',
    solverRequestName: 'Combo Pilots',
    solverStatus: 'Done-failed',
    timestamp: '27 June, 2020 at 19:02',
  },
  {
    notificationId: 6,
    scenarioName: 'Novermber 2020 Production',
    solverRequestName: 'Combo Pilots',
    solverStatus: 'Done-failed',
    timestamp: '27 June, 2020 at 19:02',
  },
];
