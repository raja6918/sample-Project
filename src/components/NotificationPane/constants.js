export const FREE = 'Free';
export const READ_ONLY = 'ReadOnly';
export const EDIT = 'Edit';
