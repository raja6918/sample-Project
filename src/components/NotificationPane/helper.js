import moment from 'moment';
import storage from '../../utils/storage';
import { t } from 'i18next';

export function clearSolverInfoInStorage() {
  storage.removeItem('solverRequestSelected');
  storage.removeItem('solverRequests');
}

export function refreshScenarioInStorage(scenario) {
  const locallyOpenedScenario = storage.getItem('openScenario') || {};
  if (locallyOpenedScenario.id !== scenario.id) {
    /* When user changes scenario, we clear any data related with solver requests*/
    clearSolverInfoInStorage();
  }
  storage.setItem('openScenario', scenario);
  storage.setItem('lastOpenedScenario', '');
}

export const timeModifier = (time = null) => {
  let modifiedTime = null;
  if (time)
    modifiedTime = moment.utc(time).format(t('NOTIFICATION.defaultTimeFormat'));
  return modifiedTime;
};
