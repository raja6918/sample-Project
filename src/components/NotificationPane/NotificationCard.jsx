import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { Delete } from '@material-ui/icons';

import SolverIcon from '../../App/Solver/SolverIcon/SolverIcon';
import { shortenText } from '../../utils/common';
import unreadIcon from './images/unreadIcon.svg';
import ReactMarkDown from 'react-markdown';
import { timeModifier } from './helper';
import SolverNotificationLink from './SolverNotificationLinkRenderer';

const textOverflowStyle = css`
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
  display: block;
`;

const CardHolder = styled.div`
  display: flex;
  flex-direction: column;
  height: 100px;
  padding: 3px 0;
  font-size: 12px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.24);
  background: #fafafa;

  & :hover {
    background: #e8f4fc;
  }
`;

const Row1 = styled.div`
  display: flex;
  flex-basis: 25%;
  align-items: center;
  padding: 5px 20px;

  & > span:nth-child(2) {
    font-weight: 600;
    margin-right: 3px;
  }
  & > span:nth-child(3) {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: space-between;
    & > span {
      width: 250px;
      ${textOverflowStyle};
    }
    & > svg {
      color: #0a75c2;
    }
  }
`;

const Row2 = styled.div`
  flex-basis: 50%;
  padding: 0 20px;

  & > div {
    width: 322px;
    & > span {
      color: #5098e7;
      cursor: pointer;
      word-break: break-all;
    }
  }
`;

const Row3 = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  flex-basis: 25%;
  padding: 5px 20px;
  & > span {
    color: #5e5e5e;
  }
  & > img {
    width: 10px;
    margin: 7px;
  }
`;

class NotificationCard extends Component {
  renderNotification = (notification, t) => {
    let jsx = null;
    switch (notification.eventType) {
      case 'solver':
        jsx = (
          <ReactMarkDown
            source={
              notification.status === 'Done-success'
                ? t('NOTIFICATION.notificationTexts.solverRequestCompleted', {
                    solverRequestName: shortenText(notification.solverName, 70),
                    solverPath: `solver/${notification.scenarioId}/${
                      notification.solverId
                    }`,
                  })
                : t('NOTIFICATION.notificationTexts.solverRequestFailed', {
                    solverRequestName: shortenText(notification.solverName, 70),
                    solverPath: `solver/${notification.scenarioId}/${
                      notification.solverId
                    }`,
                  })
            }
            renderers={{
              paragraph: 'div',
              link: SolverNotificationLink,
            }}
          />
        );

        break;
      default:
        break;
    }
    return jsx;
  };

  render() {
    const { t, deleteNotification, notificationResponse } = this.props;
    const {
      eventId,
      scenarioName,
      viewStatus,
      lastModified,
    } = this.props.notificationResponse;
    return (
      <CardHolder>
        <Row1>
          <SolverIcon
            style={{
              width: '16px',
              marginRight: '5px',
            }}
            status={notificationResponse.alertType}
          />
          <span>{t('NOTIFICATION.notificationTexts.scenario')}</span>
          <span>
            <span>{scenarioName}</span>
            <Delete
              style={{ cursor: 'pointer' }}
              onClick={() => {
                deleteNotification(eventId);
              }}
            />
          </span>
        </Row1>
        <Row2>{this.renderNotification(notificationResponse, t)}</Row2>
        <Row3>
          <span>{timeModifier(lastModified)}</span>
          {viewStatus === 'UNSEEN' && (
            <img src={unreadIcon} alt={'notification unread icon'} />
          )}
        </Row3>
      </CardHolder>
    );
  }
}

NotificationCard.propTypes = {
  t: PropTypes.func.isRequired,
  deleteNotification: PropTypes.func.isRequired,
  notificationResponse: PropTypes.shape([{}]).isRequired,
};

export default NotificationCard;
