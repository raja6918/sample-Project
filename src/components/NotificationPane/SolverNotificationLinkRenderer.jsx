import React, { useState } from 'react';
import storage from '../../utils/storage';
import history from '../../history';
import ModalLoader from '../ModalLoader';
import scenarioService from '../../services/Scenarios';
import * as solverService from '../../services/Solver';
import { t } from 'i18next';
import styled, { css } from 'styled-components';
import Notification from '../Notification';
import { closeNotificationBar } from '../../actions/generic';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { FREE, READ_ONLY, EDIT } from './constants';
import { refreshScenarioInStorage } from './helper';
import { shortenText } from '../../utils/common';
import Dialog from '../Dialog/Form';
import ReactMarkdown from 'react-markdown';

const StyledDialog = styled(Dialog)`
  .MuiButton-contained {
    width: 150px;
  }
  & .MuiDialogContent-root {
    margin-bottom: 85px;
    font-family: 'Roboto';
    font-size: 14px;
  }
  & .MuiDialogTitle-root .MuiTypography-root b {
    font-weight: 400;
  }
`;

export const SolverNotificationLinkRender = props => {
  const [openLoader, updateLoader] = useState(false);
  const [title, updateTitle] = useState(null);
  const [message, updateMessage] = useState('');
  const type = 'error';
  const [scenarioSolverToBeNavigated, setNavigateScenarioSolver] = useState({
    scenarioToNavigate: {},
    currentScenarioOpened: {},
    solverToSelect: {},
  });
  const [dialogOpen, OpenDialog] = useState(false);

  const redirectToSolver = (scenario, solverId, readOnly = false) => {
    const route = {
      pathname: `/solver/${scenario.id}`,
      state: {
        openItemId: scenario.id,
        openSolverId: solverId,
        openItemName: scenario.name,
        readOnly,
      },
    };
    history.push(route);
  };

  const openScenario = async (currentscenarioId, scenario, solverId) => {
    OpenDialog(false);
    const scenarioToBeOpened = scenario.id;
    try {
      storage.removeItem(`openTimeLineWindows`);
      storage.removeItem(`rulesetId`);
      updateLoader(true);
      updateTitle(t('SCENARIOS.process.open'));
      switch (scenario.status) {
        case FREE: {
          if (currentscenarioId)
            await scenarioService.closeScenario(currentscenarioId);
          await scenarioService.openScenario(scenarioToBeOpened);
          refreshScenarioInStorage(scenario);
          redirectToSolver(scenario, solverId);
          break;
        }
        case READ_ONLY:
          updateLoader(true);
          updateTitle(t('SCENARIOS.process.open'));
          storage.setItem('openScenario', scenario);
          redirectToSolver(scenario, solverId, true);
          break;
        case EDIT:
          updateLoader(true);
          updateTitle(t('SCENARIOS.process.open'));
          if (currentscenarioId)
            await scenarioService.closeScenario(currentscenarioId);
          refreshScenarioInStorage(scenario);
          redirectToSolver(scenario, solverId);
          break;
        default:
      }
      setTimeout(() => {
        updateLoader(false);
        props.closeNotificationBar(true);
      }, 5000);
    } catch (error) {
      updateLoader(false);
      updateMessage(t('ERRORS.SCENARIO_ID_NOT_EXIST', [scenarioToBeOpened]));
    }
  };

  const handleOk = () => {
    const {
      currentScenarioOpened,
      scenarioToNavigate,
      solverToSelect,
    } = scenarioSolverToBeNavigated;
    storage.removeItem('openScenario');
    openScenario(
      currentScenarioOpened.id,
      scenarioToNavigate,
      solverToSelect.id
    );
  };

  const handleCancel = () => {
    OpenDialog(false);
    setNavigateScenarioSolver({
      scenarioToNavigate: {},
      solverToSelect: {},
      currentScenarioOpened: {},
    });
  };

  const handleNavigate = async (e, props) => {
    let alreadyNotNavigated = true;
    const scenarioSolverArray = props.href.split('/');
    const currentScenario = storage.getItem('openScenario');
    const loggedInUser = props.userData;
    const scenarioToBeOpened = parseInt(
      scenarioSolverArray[scenarioSolverArray.length - 2],
      10
    );
    const solverToBeOpened = parseInt(
      scenarioSolverArray[scenarioSolverArray.length - 1],
      10
    );

    try {
      const [scenario, solver] = await Promise.all([
        scenarioService.getScenarioDetails(scenarioToBeOpened),
        solverService.getSolverRequestById(
          scenarioToBeOpened,
          loggedInUser.id,
          solverToBeOpened
        ),
      ]);
      if (!currentScenario) {
        //user is in no scenario
        storage.removeItem(`openTimeLineWindows`);
        storage.removeItem(`rulesetId`);
        updateLoader(true);
        updateTitle(t('SCENARIOS.process.open'));
        await scenarioService.openScenario(scenarioToBeOpened);
        switch (scenario.status) {
          case READ_ONLY: {
            storage.setItem('openScenario', scenario);
            redirectToSolver(scenario, solverToBeOpened, true);
            break;
          }
          case EDIT:
          case FREE: {
            storage.setItem('openScenario', scenario);
            redirectToSolver(scenario, solverToBeOpened);
            break;
          }
          default:
            redirectToSolver(scenario, solverToBeOpened);
        }
        updateLoader(false);
        props.closeNotificationBar(true);
        alreadyNotNavigated = false;
      }

      if (currentScenario && currentScenario.id === scenarioToBeOpened) {
        redirectToSolver(
          currentScenario,
          solverToBeOpened,
          currentScenario.status === READ_ONLY
        );
        props.closeNotificationBar(true);
        alreadyNotNavigated = false;
      }

      //different scenario
      if (alreadyNotNavigated) {
        setNavigateScenarioSolver({
          scenarioToNavigate: scenario,
          solverToSelect: solver,
          currentScenarioOpened: currentScenario,
        });
        OpenDialog(true);
      }
    } catch (err) {
      updateLoader(false);
      updateMessage(t('ERRORS.SCENARIO_ID_NOT_EXIST', [scenarioToBeOpened]));
    }
  };
  const handleClear = () => {
    updateMessage('');
  };

  return (
    <React.Fragment>
      <span
        onClick={e => {
          handleNavigate(e, props);
        }}
      >
        {props.children
          ? shortenText(props.children[0].props.value, 50)
          : t('NOTIFICATION.defaultSolverName')}
      </span>
      <ModalLoader open={openLoader} title={title} color="white" />
      <Notification message={message} type={type} clear={handleClear} />
      <StyledDialog
        open={dialogOpen}
        handleOk={handleOk}
        handleCancel={handleCancel}
        title={t('NOTIFICATION.notificationpopup.title')}
        okButton={t('NOTIFICATION.notificationpopup.OKbutton')}
        className="notification_dialog"
        onClose={handleCancel}
        renderer={true}
      >
        <ReactMarkdown
          source={t('NOTIFICATION.notificationpopup.message', {
            solverRequestName: scenarioSolverToBeNavigated.solverToSelect.name,
            scenarioName: scenarioSolverToBeNavigated.scenarioToNavigate.name,
          })}
        />
      </StyledDialog>
    </React.Fragment>
  );
};
const mapStateToProps = state => {
  return { userData: state.user.userData };
};
const mapDispatchToProps = dispatch => {
  return {
    closeNotificationBar: bool => dispatch(closeNotificationBar(bool)),
  };
};

SolverNotificationLinkRender.propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.arrayOf(PropTypes.object).isRequired,
  closeNotificationBar: PropTypes.func.isRequired,
  userData: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    email: PropTypes.string,
    role: PropTypes.string,
    userName: PropTypes.string,
  }),
};

SolverNotificationLinkRender.defaultProps = {
  userData: {},
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SolverNotificationLinkRender);
