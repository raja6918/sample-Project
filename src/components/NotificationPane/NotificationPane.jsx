import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import PerfectScrollbar from 'react-perfect-scrollbar';
import 'react-perfect-scrollbar/dist/css/styles.css';

import { DeleteSweep } from '@material-ui/icons';
import Base from '../FormDrawer/Base';
import NotificationCard from './NotificationCard';
import { perfectScrollConfig } from '../../utils/common';
import storage from '../../utils/storage';
import * as notificationService from '../../services/Notifications';
import {
  fetchNotifications,
  deleteSingleNotification,
  deleteAllNotification,
} from '../../actions/generic';
import { connect } from 'react-redux';
import debounce from 'lodash/debounce';
import Notification from '../Notification';

const FormContainer = styled.div`
  position: relative;
  height: 100vh;
  width: 400px;
`;

const FormHeader = styled.div`
  height: 113px;
  padding: 20px 16px 35px 20px;
  background: #5098e7;
  color: #ffffff;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;

  & div {
    display: flex;
    justify-content: space-between;
    align-items: flex-end;
    height: 100%;
    font-size: 18px;
    font-weight: 500;
  }
`;

const FormBody = styled.div`
  background: #f7f7f7;
  height: calc(100vh - 113px);
`;

const ClearMessage = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100%;
  color: #8c8c8c;
`;

export class NotificationPane extends Component {
  state = {
    notificationResponse: this.props.notificationResponse,
    message: '',
    snackType: 'error',
  };
  user = null;
  componentDidMount() {
    const { userData } = this.props;
    this.user = userData;
  }

  componentWillReceiveProps(newProps) {
    this.setState({ notificationResponse: newProps.notificationResponse });
  }

  deleteNotification = notificationId => {
    const { t } = this.props;
    const payload = {
      eventIds: [notificationId],
      status: 'CLEARED',
    };
    notificationService
      .updateNotificationStatus(payload)
      .then(() => {
        this.props.deleteSingleNotification(notificationId);
      })
      .catch(() => {
        this.setState({
          message: t('ERRORS.NOTIFICATIONS.message'),
        });
      });
  };

  sweepNotification = () => {
    const { t } = this.props;
    const payload = {
      status: 'CLEARED',
    };
    notificationService
      .updateAllNotifications(payload, this.user.id)
      .then(() => {
        this.props.deleteAllNotification();
      })
      .catch(() => {
        this.setState({
          message: t('ERRORS.NOTIFICATIONS.message'),
        });
      });
  };
  fetchOnScroll = debounce(
    () => this.props.fetchDataOnScroll(this.user.id, true),
    2000
  );

  render() {
    const { t, ...rest } = this.props;
    const { message, snackType } = this.state;
    const isNotificationsEmpty = this.state.notificationResponse.length === 0;
    return (
      <Base anchor="right" {...rest}>
        <FormContainer>
          <FormHeader>
            <div>
              <span>{t('NOTIFICATION.notificationPaneHeader')}</span>{' '}
              <DeleteSweep
                onClick={this.sweepNotification}
                style={
                  isNotificationsEmpty
                    ? { color: '#91b8e4' }
                    : { cursor: 'pointer' }
                }
                disabled={true}
              />
            </div>
          </FormHeader>
          <FormBody>
            <PerfectScrollbar
              option={perfectScrollConfig}
              onScrollDown={this.fetchOnScroll}
            >
              <div>
                {this.state.notificationResponse.map((notification, i) => (
                  <NotificationCard
                    t={t}
                    key={
                      notification ? notification.eventId : `notification${i}`
                    }
                    notificationResponse={notification}
                    deleteNotification={this.deleteNotification}
                  />
                ))}
              </div>
              {isNotificationsEmpty && (
                <ClearMessage>
                  {t('NOTIFICATION.notificationTexts.noNotification')}
                </ClearMessage>
              )}
            </PerfectScrollbar>
          </FormBody>
        </FormContainer>
        <Notification message={message} type={snackType} />
      </Base>
    );
  }
}

const mapStateToProps = state => {
  return {
    notificationResponse: state.notifications['notifications'],
    userData: state.user.userData,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    fetchDataOnScroll: (userId, scroll) =>
      dispatch(fetchNotifications(userId, scroll)),
    deleteSingleNotification: eventId =>
      dispatch(deleteSingleNotification(eventId)),
    deleteAllNotification: () => dispatch(deleteAllNotification()),
  };
};

NotificationPane.propTypes = {
  t: PropTypes.func.isRequired,
  notificationResponse: PropTypes.arrayOf(PropTypes.object).isRequired,
  fetchDataOnScroll: PropTypes.func.isRequired,
  deleteSingleNotification: PropTypes.func.isRequired,
  deleteAllNotification: PropTypes.func.isRequired,
  userData: PropTypes.shape({
    id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    email: PropTypes.string,
    role: PropTypes.string,
    userName: PropTypes.string,
  }),
};

NotificationPane.defaultProps = {
  userData: {},
};

const NotificationPaneComponent = connect(
  mapStateToProps,
  mapDispatchToProps
)(NotificationPane);

export default NotificationPaneComponent;
