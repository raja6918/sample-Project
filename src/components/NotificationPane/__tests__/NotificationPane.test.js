import React from 'react';
import { shallow } from 'enzyme';

import { NotificationPane } from '../NotificationPane';

const notificationResponse = [
  {
    notificationId: 1,
    scenarioName: 'July 2020 Production',
    solverRequestName: 'Combo Pilots',
    solverStatus: 'Done-success',
    timestamp: '27 June, 2020 at 19:02',
    unread: true,
  },
  {
    notificationId: 4,
    scenarioName: 'September 2020 Production',
    solverRequestName: 'Pilots 777',
    solverStatus: 'Done-success',
    timestamp: '27 June, 2020 at 19:02',
  },
];

test('NotificationPane component renders correct', () => {
  const minProps = {
    t: jest.fn(),
    isOpen: true,
    handleCancel: jest.fn(),
    notificationResponse: notificationResponse,
    fetchDataOnScroll: jest.fn(),
    deleteSingleNotification: jest.fn(),
    deleteAllNotification: jest.fn(),
  };
  const wrapper = shallow(<NotificationPane {...minProps} />);
  expect(wrapper).toMatchSnapshot();
});
