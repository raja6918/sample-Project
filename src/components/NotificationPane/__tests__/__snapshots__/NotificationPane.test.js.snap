// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationPane component renders correct 1`] = `
<Base
  anchor="right"
  deleteAllNotification={[MockFunction]}
  deleteSingleNotification={[MockFunction]}
  fetchDataOnScroll={[MockFunction]}
  handleCancel={[MockFunction]}
  isOpen={true}
  notificationResponse={
    Array [
      Object {
        "notificationId": 1,
        "scenarioName": "July 2020 Production",
        "solverRequestName": "Combo Pilots",
        "solverStatus": "Done-success",
        "timestamp": "27 June, 2020 at 19:02",
        "unread": true,
      },
      Object {
        "notificationId": 4,
        "scenarioName": "September 2020 Production",
        "solverRequestName": "Pilots 777",
        "solverStatus": "Done-success",
        "timestamp": "27 June, 2020 at 19:02",
      },
    ]
  }
  userData={Object {}}
>
  <NotificationPane__FormContainer>
    <NotificationPane__FormHeader>
      <div>
        <span />
         
        <Memo([object Object])
          disabled={true}
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        />
      </div>
    </NotificationPane__FormHeader>
    <NotificationPane__FormBody>
      <ScrollBar
        className=""
        component="div"
        containerRef={[Function]}
        onScrollDown={[Function]}
        option={
          Object {
            "swipeEasing": true,
            "wheelSpeed": 0.4,
          }
        }
      >
        <div>
          <NotificationCard
            deleteNotification={[Function]}
            notificationResponse={
              Object {
                "notificationId": 1,
                "scenarioName": "July 2020 Production",
                "solverRequestName": "Combo Pilots",
                "solverStatus": "Done-success",
                "timestamp": "27 June, 2020 at 19:02",
                "unread": true,
              }
            }
            t={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "NOTIFICATION.notificationPaneHeader",
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
          />
          <NotificationCard
            deleteNotification={[Function]}
            notificationResponse={
              Object {
                "notificationId": 4,
                "scenarioName": "September 2020 Production",
                "solverRequestName": "Pilots 777",
                "solverStatus": "Done-success",
                "timestamp": "27 June, 2020 at 19:02",
              }
            }
            t={
              [MockFunction] {
                "calls": Array [
                  Array [
                    "NOTIFICATION.notificationPaneHeader",
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
          />
        </div>
      </ScrollBar>
    </NotificationPane__FormBody>
  </NotificationPane__FormContainer>
  <Notification
    autoHideDuration={5000}
    bottomheight={0}
    buttonText=""
    clear={[Function]}
    displayCloseButton={false}
    message=""
    onLinkAction={[Function]}
    renderers={
      Object {
        "paragraph": "span",
      }
    }
    type="error"
  />
</Base>
`;
